runtime/centrifuge/src/migrations.rs

```diff
@@ -9,5 +9,127 @@
 // but WITHOUT ANY WARRANTY; without even the implied warranty of
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
+use crate::{Runtime, Weight};
 
-pub type UpgradeCentrifuge1021 = ();
+pub type UpgradeCentrifuge1021 = anemoy_pool::Migration;
+
+/// Migrate the Anemoy Pool's currency from LpEthUSC to Circle's USDC,
+/// native on Polkadot's AssetHub.
+mod anemoy_pool {
+	use cfg_primitives::PoolId;
+	use cfg_traits::PoolInspect;
+	use cfg_types::tokens::CurrencyId;
+	#[cfg(feature = "try-runtime")]
+	use codec::{Decode, Encode};
+	#[cfg(feature = "try-runtime")]
+	use frame_support::ensure;
+	use frame_support::traits::{fungibles::Inspect, OnRuntimeUpgrade};
+	#[cfg(feature = "try-runtime")]
+	use pallet_pool_system::PoolDetailsOf;
+	use sp_std::vec;
+	#[cfg(feature = "try-runtime")]
+	use sp_std::vec::Vec;
+
+	use super::*;
+	use crate::PoolSystem;
+
+	const ANEMOY_POOL_ID: PoolId = 4_139_607_887;
+	const LP_ETH_USDC: CurrencyId = CurrencyId::ForeignAsset(100_001);
+	const DOT_NATIVE_USDC: CurrencyId = CurrencyId::ForeignAsset(6);
+
+	pub struct Migration;
+
+	impl OnRuntimeUpgrade for Migration {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+			let pool_details: PoolDetailsOf<Runtime> =
+				PoolSystem::pool(ANEMOY_POOL_ID).ok_or("Could not find Anemoy Pool")?;
+
+			ensure!(
+				pool_details.currency == LP_ETH_USDC,
+				"anemoy_pool::Migration: pre_upgrade failing as Anemoy's currency should be LpEthUSDC"
+			);
+
+			Ok(pool_details.encode())
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			// To be executed at 1021, reject higher spec_versions
+			if crate::VERSION.spec_version >= 1022 {
+				log::error!(
+					"anemoy_pool::Migration: NOT execution since VERSION.spec_version >= 1022"
+				);
+				return Weight::zero();
+			}
+
+			let (sanity_checks, weight) = verify_sanity_checks();
+			if !sanity_checks {
+				log::error!("anemoy_pool::Migration: Sanity checks FAILED");
+				return weight;
+			}
+
+			pallet_pool_system::Pool::<Runtime>::mutate(ANEMOY_POOL_ID, |details| {
+				let details = details.as_mut().unwrap();
+				details.currency = DOT_NATIVE_USDC;
+				log::info!("anemoy_pool::Migration: currency set to USDC ✓");
+			});
+
+			weight.saturating_add(
+				<Runtime as frame_system::Config>::DbWeight::get().reads(
+					pallet_pool_system::Pool::<Runtime>::iter_keys()
+						.count()
+						.saturating_mul(2) as u64,
+				),
+			)
```

Comment:
Why are we adding two reads multiplied with the number of pool keys? Shouldn't it just add single read and a write?

Reply:
fixed. I guess the result of moving coding around and changes to the implementation. Made this add 1 read and 1 write here, for the pool lookup and mutate, respectively. thanks!